modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/AuthManager.java

```diff
@@ -60,13 +65,23 @@
      */
     private Authenticator authenticator;
 
+    private Authorizer authorizer;
+
     public AuthManager(StartupContext startupContext) throws Exception {
         BrokerConfigProvider configProvider = startupContext.getService(BrokerConfigProvider.class);
         brokerAuthConfiguration = configProvider
                 .getConfigurationObject(BrokerAuthConfiguration.NAMESPACE, BrokerAuthConfiguration.class);
+        BrokerCommonConfiguration commonConfigs
+                = configProvider.getConfigurationObject(BrokerCommonConfiguration.NAMESPACE,
+                                                        BrokerCommonConfiguration.class);
         startupContext.registerService(AuthManager.class, this);
         authenticator = new AuthenticatorFactory().getAuthenticator(startupContext,
                                                                     brokerAuthConfiguration.getAuthentication());
+        AuthProvider authProvider = new AuthProviderFactory().getAuthorizer(commonConfigs,
+                                                                            brokerAuthConfiguration,
+                                                                            startupContext);
+        authorizer = new AuthorizerFactory().getAutStore(authProvider, commonConfigs,
+                                                         brokerAuthConfiguration, startupContext);
```

Comment:
Shall we assert here in the constructor if the user has configured correctly?

- authentication(enabled), authorization(enabled) - allowed 
- authentication(disabled), authorization(enabled) - not allowed 
- authentication(enabled), authorization(disabled) - allowed 
- authentication(disabled), authorization(disabled) - allowed

Reply:
Fixed in d1ed2c6